cmd/ipfs/init.go

```diff
@@ -49,19 +48,17 @@ var initCmd = &cmds.Command{
 			res.SetError(err, cmds.ErrNormal)
 			return
 		}
+
 		if !bitsOptFound {
 			nBitsForKeypair = nBitsForKeypairDefault
 		}
 
-		rpipe, wpipe := io.Pipe()
-		go func() {
-			defer wpipe.Close()
-			if err := doInit(wpipe, req.Context().ConfigRoot, force, nBitsForKeypair); err != nil {
-				res.SetError(err, cmds.ErrNormal)
-				return
-			}
-		}()
-		res.SetOutput(rpipe)
+		var buf bytes.Buffer
+		if err := doInit(&buf, req.Context().ConfigRoot, force, nBitsForKeypair); err != nil {
+			res.SetError(err, cmds.ErrNormal)
+			return
+		}
+		res.SetOutput(&buf)
```

Comment:
Im serious on that one, lol. its cleaner to initialize the variable as a pointer than it is to make it a direct object and then give everything its address. It also makes no difference (escape analysis throws the whole 'put it on the stack' concept out the window).

Oh and programmers only use tabs. spaces are for barbarians and depraved individuals who write python.